aten/src/ATen/core/ivalue.cpp

```diff
@@ -60,6 +60,8 @@ TypePtr IValue::type() const {
       return DeviceObjType::get();
     case Tag::Object:
       return toObjectRef().type();
+    case Tag::PythonObject:
```

Comment:
The tag is PythonObject but the name of the type is PythonObj? Futhermore, python calls these things PyObject and pybind11 its py::object? Can both of these things be renamed to PyObject please?

Reply:
This is primarily due to name collisions, I can rename `PythonObjType` to `PyObjectType` in jit_type for sure. but in IValue, we need to deal with name collisions between `ivalue::PyObject` and `PyObject` itself.

Reply:
I think I can keep the tag with `Tag::PyObject`, but with `ivalue::PyObjectHolder` be the real holding class.

Reply:
one is ivalue::Tag::PyObject and the other is ivalue::PyObject, right? Unless I am missing something, it doesn't seem like those conflict.

Reply:
I did another try on this. the name collision happens between `ivalue::PyObject` and our global python stub `PyObject` defined in https://github.com/pytorch/pytorch/blob/master/c10/util/python_stub.h#L4. C++ compiler messed up with symbol overriding with global namespace.

So the changes I made is that I keep the ivalue class to be `PyObjectHolder`, all other things still be `PyObject`.